core/src/main/java/io/kestra/core/runners/Worker.java

```diff
@@ -308,7 +308,8 @@ private WorkerTaskResult run(WorkerTask workerTask, Boolean cleanUp) throws Queu
         WorkerTask finalWorkerTask = Failsafe
             .with(AbstractRetry.<WorkerTask>retryPolicy(workerTask.getTask().getRetry())
                 .handleResultIf(result -> result.getTaskRun().lastAttempt() != null &&
-                    Objects.requireNonNull(result.getTaskRun().lastAttempt()).getState().getCurrent() == State.Type.FAILED
+                        result.getTaskRun().lastAttempt().getState().getCurrent() == State.Type.FAILED &&
```

Comment:
It lacks a test to cover it I think

Reply:
I can try to make one but as it's by nature racy the test might be flaky.